lib/checks/tables/td-headers-attr.js

```diff
@@ -1,40 +1,36 @@
-var cells = [];
-
-for (
-	var rowIndex = 0, rowLength = node.rows.length;
-	rowIndex < rowLength;
-	rowIndex++
-) {
-	var row = node.rows[rowIndex];
-	for (
-		var cellIndex = 0, cellLength = row.cells.length;
-		cellIndex < cellLength;
-		cellIndex++
-	) {
+const cells = [];
+const reviewCells = [];
+const badCells = [];
+
+for (let rowIndex = 0; rowIndex < node.rows.length; rowIndex++) {
+	const row = node.rows[rowIndex];
+
+	for (let cellIndex = 0; cellIndex < row.cells.length; cellIndex++) {
 		cells.push(row.cells[cellIndex]);
 	}
 }
 
-var ids = cells.reduce(function(ids, cell) {
+const ids = cells.reduce((ids, cell) => {
 	if (cell.getAttribute('id')) {
 		ids.push(cell.getAttribute('id'));
 	}
 	return ids;
 }, []);
 
-var badCells = cells.reduce(function(badCells, cell) {
-	var isSelf, notOfTable;
+cells.forEach(cell => {
+	let isSelf, notOfTable;
```

Comment:
Lets give these the defaults we expect them to have, instead of relying on type casting of undefined.
```suggestion
	let isSelf = false;
	let notOfTable = false;
```